proto/osmosis/gamm/v1beta1/genesis.proto

```diff
@@ -3,25 +3,35 @@ package osmosis.gamm;
 
 import "gogoproto/gogo.proto";
 import "google/protobuf/any.proto";
+import "google/protobuf/duration.proto";
 import "cosmos_proto/cosmos.proto";
 import "cosmos/base/v1beta1/coin.proto";
 
 // Params holds parameters for the incentives module
 message Params {
-    repeated cosmos.base.v1beta1.Coin pool_creation_fee = 1 [
-        (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
-        (gogoproto.moretags) = "yaml:\"pool_creation_fee\"",
-        (gogoproto.nullable) = false
-    ];
+  repeated cosmos.base.v1beta1.Coin pool_creation_fee = 1 [
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (gogoproto.moretags) = "yaml:\"pool_creation_fee\"",
+    (gogoproto.nullable) = false
+  ];
+  uint64 num_twap_history_per_deletion = 2;
+  // interval is in the unit of block height
+  uint64 twap_history_deletion_interval = 3;
```

Comment:
Oh interesting, I guess deletions are happening in begin/end block?

Reply:
yep the deletion happens on the end block!